atomic_reactor/plugins/exit_koji_import.py

```diff
@@ -0,0 +1,280 @@
+            'start_time': start_time,
+            'end_time': int(time.time()),
+            'extra': extra,
+        }
+
+        if self.metadata_only:
+            build['metadata_only'] = True
+
+        return build
+
+    def combine_metadata_fragments(self):
+        try:
+            metadata = get_build_json()["metadata"]
+            self.build_id = metadata["name"]
+        except KeyError:
+            self.log.error("No build metadata")
+            raise
+
+        metadata_version = 0
+
+        build = self.get_build(metadata)
+        """
+        # get all the platform names
+        build_result = self.workflow.build_result
+        annotations = build_result.annotations
+        worker_builds = annotations['worker-builds']
+        platforms = worker_builds.items()
+        """
+
+        metadatas = self.workflow.postbuild_results.get(FetchWorkerMetadataPlugin.key)
+        buildroot = self.get_buildroot(metadatas)
+        output = self.get_output(metadatas)
+
+        koji_metadata = {
+            'metadata_version': metadata_version,
+            'build': build,
+            'buildroots': [buildroot],
+            'output': [output],
+        }
+        return koji_metadata
+
+    def login(self):
+        """
+        Log in to koji
+
+        :return: koji.ClientSession instance, logged in
+        """
+
+        # krbV python library throws an error if these are unicode
+        auth_info = {
+            "proxyuser": self.koji_proxy_user,
+            "ssl_certs_dir": self.koji_ssl_certs,
+            "krb_principal": str(self.koji_principal),
+            "krb_keytab": str(self.koji_keytab)
+        }
+        return create_koji_session(str(self.kojihub), auth_info)
+
+    def run(self):
+        """
+        Run the plugin.
+        """
+
+        if ((self.koji_principal and not self.koji_keytab) or
+                (self.koji_keytab and not self.koji_principal)):
+            raise RuntimeError("specify both koji_principal and koji_keytab "
+                               "or neither")
+
+        # Only run if the build was successful
+        if self.workflow.build_process_failed:
+            self.log.info("Not importing failed build to koji")
+            return
+
+        session = self.login()
+
+        server_dir = get_koji_upload_dir(self.workflow)
+
+        koji_metadata = self.combine_metadata_fragments()
+
+        try:
+            build_info = session.CGImport(koji_metadata, server_dir)
+        except Exception:
+            self.log.debug("metadata: %r", koji_metadata)
+            raise
+
+        # Older versions of CGImport do not return a value.
+        build_id = build_info.get("id") if build_info else None
+
+        self.log.debug("Build information: %s",
+                       json.dumps(build_info, sort_keys=True, indent=4))
+
+        # If configured, koji_tag_build plugin will perform build tagging
+        tag_later = are_plugins_in_order(self.workflow.exit_plugins_conf,
```

Comment:
I think this code exists in `koji_import` for backwards compatibility with plugin configurations that do not yet use `koji_tag_build` plugin. Can we assume that if someone is using this new plugin, they have already switched over to using `koji_tag_build`? I'm suggesting we start this new plugin without tagging support.